src/storage/txn/actions/prewrite.rs

```diff
@@ -59,7 +58,14 @@ pub fn prewrite<S: Snapshot>(
     };
 
     if mutation.should_not_write {
-        return Ok((TimeStamp::zero(), OldValue::Unspecified));
+        let min_commit_ts = if mutation.need_min_commit_ts() {
+            // Don't calculate the min_commit_ts according to the concurrency manager's max_ts
+            // for a should_not_write mutation because it's not persisted and doesn't change data.
+            cmp::max(txn_props.min_commit_ts, txn_props.start_ts.next())
```

Comment:
Do we need to also maximize it with `for_update_ts.next()`?

Reply:
No.
```rust
        let should_not_write = mutation.should_not_write();
        if txn_props.is_pessimistic() && should_not_write {
            return Err(box_err!(
                "cannot handle checkNotExists in pessimistic prewrite"
            ));
        }
```